### PR TITLE
Feature/save in extended data

### PR DESCRIPTION
**Pull request by AyakorK**
Hello ! In this PR you will find 

- Saves of new registration fields
- Add of specs to test this new feature
- Update of README with some screenshots.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,37 @@ And then execute:
 bundle
 ```
 
+## How to use it
```
Comment (Quentinchampenois) [inline README.md:28]: I was thinking about just displaying a screenshot of the BO form and the signup form just to have an overview of changes of the module

File: app/commands/decidim/create_registration.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        nickname: form.nickname,
+        password: form.password,
+        password_confirmation: form.password_confirmation,
+        organization: form.current_organization,
+        tos_agreement: form.tos_agreement,
+        newsletter_notifications_at: form.newsletter_at,
+        email_on_notification: true,
+        accepted_tos_version: form.current_organization.tos_version,
+        locale: form.current_locale,
+        extended_data: {
```
Comment (Quentinchampenois) [inline app/commands/decidim/create_registration.rb]: Nice override, can you try to remove this file and create a new extend as follow : 
```suggestion
        extended_data: extended_data
        }
        
        def extended_data
         {
           birth_date: form.birth_date,
           minimum_age: form.minimum_age
         }
        end
```

It will be easier to update for future changes

File: spec/commands/decidim/create_registration_spec.rb [hunk 3]
```diff
@@ -0,0 +1,263 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  module Comments
+    describe CreateRegistration do
+      describe "call without registration_fields" do
+        # Create an organisation with registration fields disabled
+        let(:organization) { create(:organization, registration_fields: { "enabled" => false }) }
```
Comment (Quentinchampenois) [inline spec/commands/decidim/create_registration_spec.rb]: Prefer to use available factory trait 

```suggestion
        let(:organization) { create(:organization, :registration_fields_disabled) }
```

File: spec/commands/decidim/create_registration_spec.rb [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+          describe "when user keeps the newsletter unchecked" do
+            let(:newsletter) { "0" }
+
+            it "creates a user with no newsletter notifications" do
+              expect do
+                command.call
+                expect(User.last.newsletter_notifications_at).to eq(nil)
+              end.to change(User, :count).by(1)
+            end
+          end
```
Comment (Quentinchampenois) [inline spec/commands/decidim/create_registration_spec.rb:114]: Prefer define spec for registration fields here

describe -> registration fields enabled

context -> when birth_date is mandatory but empty
-> expect to fail
context -> when minimum_age is mandatory but empty
-> expect to fail 

No need to test for the both expectation empty since there is at least one error

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
Nice work, just a need to remove the command override and create an extend at `lib/decidim/extends/commands/create_registration_extend.rb` since it is easier to maintain in future

Don't forget to require the command in the `engine.rb`

**Comment by AyakorK** [comment]
This should be good @Quentinchampenois 
I let you check and if you're looking for other changes tell me
